src/Centreon/Infrastructure/Engine/EngineConfigurationRepositoryRDB.php

```diff
@@ -85,6 +90,32 @@ public function findEngineConfigurationByHost(Host $host): ?EngineConfiguration
         return null;
     }
 
+    /**
+     * {@inheritDoc}
+     * @throws \PDOException
+     */
+    public function findEngineConfigurationByMonitoringServerId(int $monitoringServerId): ?EngineConfiguration
+    {
+        $statement = $this->db->prepare(
+            $this->translateDbName(
+                'SELECT * FROM `:db`.cfg_nagios cfg
```

Comment:
IMHO, it's better to select only needed fields, but this is not a blocking point